wots.c

```diff
@@ -129,6 +203,79 @@ void wots_pkgen(const xmss_params *params,
  * Takes a n-byte message and the 32-byte seed for the private key to compute a
  * signature that is placed at 'sig'.
  */
+typedef struct wots_sign_args
+{
+    const xmss_params *params;
+    uint32_t *addr;
+    const unsigned char *pub_seed;
+    unsigned char *sig;
+    uint16_t padding16;
+    int *lengths;
+    uint32_t start;
+    uint32_t end;
+} wots_sign_args_t;
+
+static void *wots_sign_sub(void *vars)
+{
+    wots_sign_args_t *args = vars;
+
+    for (uint32_t i = args->start; i < args->end; i++)
+    {
+        set_chain_addr(args->addr, i);
+        gen_chain(args->params, args->sig + i * args->params->n, args->sig + i * args->params->n,
+                  0, args->lengths[i], args->pub_seed, args->addr);
+    }
+    return NULL;
+}
+
+void wots_sign_mp(const xmss_params *params,
+               unsigned char *sig, const unsigned char *msg,
+               const unsigned char *seed, const unsigned char *pub_seed,
+               uint32_t addr[8])
+{
+    int lengths[params->wots_len];
+    chain_lengths(params, lengths, msg);
+    /* The WOTS+ private key is derived from the seed. */
+    expand_seed(params, sig, seed);
+
+    const uint32_t block = params->wots_len / NUM_CORES;
+    const uint32_t remain = (params->wots_len % NUM_CORES);
+    pthread_t thread[NUM_CORES + 1];
```

Comment:
I would call this: `const int thread_number = NUM_CORES + 1` so you don't repeat yourself.

Reply:
fix in https://github.com/kj3moraes/further-testing-xmss/pull/5/commits/46648c53b9aa817fef5cde8cdf7c1d2004a92b05